client/router_update.go

```diff
@@ -114,6 +115,7 @@ func (cli *VanClient) RouterUpdateVersionInNamespace(ctx context.Context, hup bo
 		substituteFlowCollector = utils.LessRecentThanVersion(originalVersion, "1.3.0")
 		addPrometheusServer = utils.LessRecentThanVersion(originalVersion, "1.4.0")
 		moveClaims = utils.LessRecentThanVersion(originalVersion, "1.5.0")
+		updateSecretsFor102 = utils.LessRecentThanVersion(originalVersion, "1.0.2")
```

Comment:
I am not sure this will be enough. In example, if a site was updated from 1.0.1
to 1.0.2, the connect.json will still have the incorrect host name. So eventually
newer versions (>=1.0.2) might have that same issue as well.

I believe to fix this, we might need an excluvise validation inside the connect.json
from skupper-local-client secret to detect whether the host name is still incorrectly
pointing to `skupper-local-client.<namespace>.svc.cluster.local`, if so, then the
`host` key within the connect.json needs to be updated.

The skupper-router-local TLS certificate itself, does not need to be recreated (I suppose),
as the CN is already being updated.

Reply:
Though I suspect the likelihood of hitting this for anyone who cares is low (and could be mitigated by a manual fix), it might be as clear and easy to just make the check as Fernando suggests.

Reply:
>  if so, then the host key within the connect.json needs to be updated.

@fgiorgetti does the host key need to be updated to an empty value, like it happens for older versions than 0.5?

Reply:
updated in dc5caa56e104f08bc63868a1ac55c0b778a5b3ce